### PR TITLE
CORS-3698: Introduce default configuration step on AWS

### PR DESCRIPTION
**Pull request by yunjiang29**
All of install-config on AWS includes many of optional configurations (see step [ipi-conf-aws](https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-ref.yaml)), like zones, instance types, in some cases, the test requires a _default_ configuration, to cover cases like [OCPBUGS-43453](https://issues.redhat.com/browse/OCPBUGS-43453), hence, `ipi-conf-aws-default` step is introduced, which will generate install-config like:
```
apiVersion: v1
metadata:
  name: ci-op-cxhzw500-2fe85
baseDomain: qe.devcluster.openshift.com
platform:
  aws:
    region: us-east-1
    userTags:
      ...
controlPlane:
  architecture: arm64
  name: master
  replicas: 3
  platform: {}
compute:
- architecture: arm64
  name: worker
  replicas: 3
  platform: {}
```

### CODE REVIEW

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-default-mini-perm-arm-f7

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-default-mini-perm-arm-f7

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-default-mini-perm-arm-f7

**Comment by yunjiang29** [comment]
test get passed:
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/59099/rehearse-59099-periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-default-mini-perm-arm-f7/1863765961295794176

**Comment by yunjiang29** [comment]
@jianlinliu @gpei PTAL

File: ci-operator/step-registry/ipi/conf/aws/default/ipi-conf-aws-default-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      (default: 3)
+  - name: CONTROL_PLANE_REPLICAS
+    default: "3"
+    documentation: |-
+      The number of control plane nodes to create for the cluster. Useful for SNO cluster provision.
+  - name: OCP_ARCH
+    default: "amd64"
+    documentation: |-
+      The architecture of the control plane nodes (e.g., amd64, arm64).
+  - name: COMPUTE_ARCH
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/aws/default/ipi-conf-aws-default-ref.yaml:31]: Seem like we do not need this ENV.

File: ci-operator/step-registry/ipi/conf/aws/default/ipi-conf-aws-default-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    default: "amd64"
+    documentation: |-
+      Compute node architecture specification.  Used for multiarch compute clusters.
+  - name: CONTROL_ARCH
+    default: "amd64"
+    documentation: |-
+      Control plane node architecture specification.  Used for multiarch compute clusters.
+  - name: USER_TAGS
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/aws/default/ipi-conf-aws-default-ref.yaml:39]: I guess we also do not need this ENV in default install workflow

**Comment by gpei** [comment]
I'm thinking if it could be possible to add a ipi-conf-default step that could be used for configuring the default options for AWS/GCP/Azure platforms, so we can easily add such job for all the three platforms?

File: ci-operator/step-registry/ipi/conf/aws/default/ipi-conf-aws-default-ref.yaml (hunk 1 above)
Comment (yunjiang29) [reply ci-operator/step-registry/ipi/conf/aws/default/ipi-conf-aws-default-ref.yaml:31]: Use single `OCP_ARCH` to control master and worker arch?

File: ci-operator/step-registry/ipi/conf/aws/default/ipi-conf-aws-default-ref.yaml (hunk 2 above)
Comment (yunjiang29) [reply ci-operator/step-registry/ipi/conf/aws/default/ipi-conf-aws-default-ref.yaml:39]: ah, right.

**Comment by yunjiang29** [comment]
> that couldn't be used for other platforms.

I just noticed this PR could be a part of https://issues.redhat.com/browse/CORS-3698, it looks like we need a similar step on Azure and GCP as well

File: ci-operator/step-registry/ipi/conf/aws/default/ipi-conf-aws-default-ref.yaml (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/aws/default/ipi-conf-aws-default-ref.yaml]: Because whatever `COMPUTE_ARCH` is set, it is always overwritten.

Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/aws/default/ipi-conf-aws-default-ref.yaml]: Okay, now `OCP_ARCH` controls master and worker arch, I am okay with it.

**Comment by gpei** [comment]
> it looks like we need a similar step on Azure and GCP as well

yes indeed, let me create a QE task to track this.

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.18-aws-ipi-default-mini-perm-arm-f7

**Pull request merged**